aten/src/ATen/native/Convolution.cpp

```diff
@@ -1139,4 +1139,22 @@ std::tuple<Tensor,Tensor,Tensor> _convolution_double_backward(
   return std::tuple<Tensor,Tensor,Tensor>{ggO, gI, gW};
 }
 
+// Performs convolution with a scaled weight, unscales, and returns the
+// unscaled results. This is done for QAT Conv-BN fusion: it contains
+// the logical part of QAT Conv-BN which is after observation (all inputs
+// are observed), and before BN.
+at::Tensor qat_conv2d_and_unscale(
+    const Tensor& input, const Tensor& scaled_weight,
+    const Tensor& scale_factor, const Tensor& bias, IntArrayRef stride,
+    IntArrayRef padding, IntArrayRef dilation, int64_t groups) {
+  Tensor result = at::convolution(
+      input, scaled_weight, bias, stride, padding, dilation, false, {{0, 0}},
+      groups);
+  result = result / scale_factor.reshape({1, -1, 1, 1});
```

Comment:
also should we make this work for conv3d as well?

Reply:
btw fusion is already extended to work with conv3d: https://github.com/pytorch/pytorch/blob/master/torch/nn/utils/fusion.py#L17

Reply:
yep, for sure.  Planning to do that in future PR, to keep PRs smaller.  Will add a TODO.